components/webext-storage/src/api.rs

```diff
@@ -77,33 +107,21 @@ pub struct StorageValueChange {
     new_value: Option<String>,
 }
 
-// This is, largely, a helper so that this serializes correctly as per the
-// chrome.storage.sync spec. If not for custom serialization it should just
-// be a plain vec
+/// A newtype for a list of changed keys and values, used to notify
+/// `storage.onChanged` listeners when an item is updated or removed. It
+/// dereferences to its contents, has a custom serializer to serialize the
+/// changes as a JSON object instead of a list, and implements
+/// `ffi_support::IntoFfi` so that it can be passed over the FFI.
 #[derive(Debug, Clone, PartialEq)]
-pub struct StorageChanges {
-    changes: Vec<StorageValueChange>,
-}
+pub struct StorageChanges(Vec<StorageValueChange>);
```

Comment:
It was fine with `changes` before, but I liked the symmetry, and the older comment saying "it should just be a plain vec" if not for the trait implementations (`Serialize`, and now `IntoFfi`) was practically screaming "newtype!" 😁

Reply:
At some point moving these related types into their own module probably makes sense - I'm not bothered if that time is now or not.